org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/marker/GradleErrorMarker.java

```diff
@@ -45,50 +45,65 @@ public static boolean belongsToBuild(IMarker marker, InternalGradleBuild build)
             if (lineNumber >= 0) {
-                marker.setAttribute(IMarker.LINE_NUMBER, lineNumber);
+                try {
+                    marker.setAttribute(IMarker.LINE_NUMBER, lineNumber);
+                } catch (CoreException e) {
+                    throw new RuntimeException(e);
+                }
             }
+        });
+    }
 
-            marker.setAttribute(IMarker.MESSAGE, trimMarkerProperty(message));
-            marker.setAttribute(IMarker.PRIORITY, IMarker.PRIORITY_HIGH);
-            marker.setAttribute(IMarker.SEVERITY, severity);
-            marker.setAttribute(ATTRIBUTE_ROOT_DIR, gradleBuild.getBuildConfig().getRootProjectDirectory().getAbsolutePath());
-            if (exception != null) {
-                String stackTrace = Throwables.getStackTraceAsString(exception);
-                marker.setAttribute(GradleErrorMarker.ATTRIBUTE_STACKTRACE, trimMarkerProperty(stackTrace));
+    private static void createMarker(IResource resource, int severity, InternalGradleBuild gradleBuild, String message, String exception, Consumer<IMarker> customMarkerConfiguration) {
+        createMarker(resource, new Consumer<IMarker>() {
+
+            @Override
+            public void accept(IMarker marker) {
+                try {
+                    marker.setAttribute(IMarker.PRIORITY, IMarker.PRIORITY_HIGH);
+                    marker.setAttribute(IMarker.SEVERITY, severity);
+                    marker.setAttribute(ATTRIBUTE_ROOT_DIR, gradleBuild.getBuildConfig().getRootProjectDirectory().getAbsolutePath());
+                    marker.setAttribute(IMarker.MESSAGE, trimMarkerProperty(message));
+                    if (exception != null) {
+                        marker.setAttribute(GradleErrorMarker.ATTRIBUTE_STACKTRACE, exception);
+                    }
+                } catch (CoreException e) {
+                    CorePlugin.logger().warn("Cannot create Gradle error marker", e);
+                    throw new RuntimeException(e);
+                }
             }
-            if (category != null) {
-                marker.setAttribute(ATTRIBUTE_PROBLEM_CATEGORY, category);
-            }
-            if (solutions != null) {
-                String solutionsString = solutions.stream().collect(Collectors.joining(System.getProperty("line.separator")));
-                marker.setAttribute(ATTRIBUTE_PROBLEM_SOLUTIONS, solutionsString);
-            }
-            if (documentationLink != null) {
-                marker.setAttribute(ATTRIBUTE_DOCUMENTATION_LINK, documentationLink);
-            }
-        } catch (CoreException e) {
+        }.andThen(customMarkerConfiguration));
+    }
+
+    private static void createMarker(IResource resource, Consumer<IMarker> markerConfiguration) {
+        try {
+            IMarker marker = resource.createMarker(GradleErrorMarker.ID);
+            markerConfiguration.accept(marker);
+        } catch (Exception e) {
```

Comment:
❓ also `CoreException` here? I don't see anything throwing a plain `Exception`

Reply:
`IResource.createMarker()`  throws `CoreException`.

Reply:
markerConfiguration catches CoreException and re-throws them as runtime exceptions.